cli/gen_runtime_vectors/ledger-paratime-transactions.md

```diff
@@ -0,0 +1,648 @@
+| INS        | byte (1)       | Instruction ID         | 0x04           |
+| P1         | byte (1)       | Request User confirmation | No = 0      |
+| P2         | byte (1)       | Parameter 2            | ignored        |
+| L          | byte (1)       | Bytes in payload       | (depends)      |
+| Path[0]    | byte (4)       | Derivation Path Data   | 44             |
+| Path[1]    | byte (4)       | Derivation Path Data   | 0              |
+| Path[2]    | byte (4)       | Derivation Path Data   | ?              |
+| Path[3]    | byte (4)       | Derivation Path Data   | ?              |
+| Path[4]    | byte (4)       | Derivation Path Data   | ?              |
+
+First three items in the derivation path will be hardened automatically hardened
+
+#### Response
+
+| Field   | Type      | Content               | Note                     |
+| ------- | --------- | --------------------- | ------------------------ |
+| PK      | byte (32) | Public Key            |                          |
+| ADDR    | byte (??) | Hex addr              |                          |
+| SW1-SW2 | byte (2)  | Return code           | see list of return codes |
+
+### SIGN_PT_ED25519
+
+#### Command
+
+| Field | Type     | Content                | Expected  |
+| ----- | -------- | ---------------------- | --------- |
+| CLA   | byte (1) | Application Identifier | 0x05      |
+| INS   | byte (1) | Instruction ID         | 0x05      |
+| P1    | byte (1) | Payload desc           | 0 = init  |
+|       |          |                        | 1 = add   |
+|       |          |                        | 2 = last  |
+| P2    | byte (1) | ----                   | not used  |
+| L     | byte (1) | Bytes in payload       | (depends) |
+
+The first packet/chunk includes only the derivation path
+
+All other packets/chunks should contain message to sign
+
+*First Packet*
+
+| Field      | Type     | Content                | Expected  |
+| ---------- | -------- | ---------------------- | --------- |
+| Path[0]    | byte (4) | Derivation Path Data   | 44        |
+| Path[1]    | byte (4) | Derivation Path Data   | 474       |
+| Path[2]    | byte (4) | Derivation Path Data   | ?         |
+| Path[3]    | byte (4) | Derivation Path Data   | ?         |
+| Path[4]    | byte (4) | Derivation Path Data   | ?         |
+
+*Other Chunks/Packets*
+
+| Field   | Type     | Content              | Expected |
+| ------- | -------- | -------------------- | -------- |
+| Data    | bytes... | Context+Message+Meta |          |
+
+Data is defined as:
+
+| Field   | Type     | Content                    | Expected     |
+| ------- | -------- | -------------------------- | ------------ |
+| CtxLen  | byte     | Context Length             |              |
+| Context | bytes..  | Context                    | CtxLen bytes |
+| Message | bytes..  | CBOR data to sign          |              |
+| Meta    | bytes..  | CBOR metadata to verify    |              |
```

Comment:
CBOR is self-contained so no size is needed. It's not O(1) time though and I'm not sure, if zondax's cbor implementation have this implemented - they certainly don't use it AFAIK.

Reply:
After a thought, I switched Message, because Meta is a simple key-value store which size is easy to compute and it may also contain some information on how to parse the subsequent Message.